php-frontend/src/main/java/org/sonar/plugins/php/api/tree/declaration/ClassDeclarationTree.java

```diff
@@ -52,8 +52,7 @@ public interface ClassDeclarationTree extends StatementTree, ClassTree {
   /**
    * Either {@link PHPKeyword#ABSTRACT abstract} or {@link PHPKeyword#FINAL final}
    */
-  @Nullable
-  SyntaxToken modifierToken();
```

Comment:
Unfortunately, we can not simply remove a function from an API class. So we have to keep this method with a deprecation annotation and a comment which recommends using the new method instead. This method must be used until the next but one LTS and behave as before. That means it may only return the abstract or final token, as well as null.